snuba/cleanup.py

```diff
@@ -47,21 +47,25 @@ def get_active_partitions(
     return [util.decode_part_str(part, part_format) for part, in response.results]
 
 
-def filter_stale_partitions(
-    parts: Sequence[util.Part], as_of: Optional[datetime] = None
-) -> Sequence[util.Part]:
-    """Filter partitions down to ones that are out of the retention
-    window based on `as_of` (default: now)."""
+def current_time() -> datetime:
+    """Returns the current day. Separate function for mocking purposes."""
+    return datetime.utcnow()
```

Comment:
Just my $0.02 but I think the way it was written earlier (passing this date as an optional argument to `filter_stale_partitions` so it can be tested more straightforward than making this a separate function then needing to do Mock in tests. Any particular reason for the change?

Reply:
I have an opposing opinion. I don't like the pattern of having parameters in a function that exist purely for tests. `as_of` was just a confusing extra parameter if you didn't know it was for test mocking. I don't think main code should be written to accomodate tests if it can be avoided.